client/client.go

```diff
@@ -453,6 +453,12 @@ type (
 	// WARNING: Worker versioning is currently experimental.
 	WorkerVersioningRules = internal.WorkerVersioningRules
 
+	// WorkflowUpdateRPCTimeoutOrCancelledError is an error that occurs when an update RPC call times out or is cancelled.
+	//
+	// Note, this is not related to any general concept of timing out or cancelling a running update, this is only related to the client call itself.
+	// NOTE: Experimental
+	WorkflowUpdateRPCTimeoutOrCancelledError = internal.WorkflowUpdateRPCTimeoutOrCancelledError
```

Comment:
I used `RPC` because it's `RPCError` in Python to differentiate between RPC and non-RPC errors (e.g. if we have an actual client-side update timeout error which is different than this one). But Go uses the `serviceerror` package to differentiate RPC errors. So if you want, you could change this to `WorkflowUpdateServiceTimeoutOrCancelledError` or something like that, unsure. But I'm fine with this too.